src/code_e0b30_len_b80.c

```diff
@@ -1,38 +1,215 @@
 #include "common.h"
+#include "map.h"
 
-INCLUDE_ASM(s32, "code_e0b30_len_b80", get_default_variation_for_song);
+void transition_music_volume_to(s16 volume);
 
-INCLUDE_ASM(s32, "code_e0b30_len_b80", func_8014A498);
+/// If the given song ID is present in gSongsUsingVariationFlag, returns the current
+/// map's `flags2 & 1` value. Otherwise, returns -1.
+///
+/// @see gSongsUsingVariationFlag
+/// @returns -1: no override; 0: override to variation 0; 1 override to variation 1
+s32 get_song_variation_override_for_cur_map(SongID songID) {
+    u32 i = 0;
+    Area* areas = gAreas;
+    SongID* allowed = gSongsUsingVariationFlag;
+
+    for (i = 0; i < ARRAY_COUNT(gSongsUsingVariationFlag); i++) {
+        if (allowed[i] == songID) {
+            Map* map = &areas[GAME_STATUS->areaID].maps[GAME_STATUS->mapID];
+
+            return map->flags2 & 1;
+        }
+    }
+
+    return -1;
+}
+
+void func_8014A498(void) {
+    MusicPlayer* src; // TODO: remove this temp when .data section for this file (incl. D_8014F6F0) is set up
+    s32 i;
+
+    for (i = 0; i < 2; i++) {
+        MusicPlayer* musicPlayers = gMusicPlayers;
+        musicPlayers[i] = *(src = &D_8014F6F0);
+    }
+
+    gMusicTargetVolume = 8;
+    gMusicUnkVolume2 = 8;
+    gMusicCurrentVolume = 8;
+    func_800561A4(8);
+}
 
 void func_8014A52C(void) {
-    D_80159AE2 = 0x08;
-    D_80159AE4 = 0x08;
+    gMusicTargetVolume = 8;
+    gMusicUnkVolume2 = 8;
 }
 
 INCLUDE_ASM(s32, "code_e0b30_len_b80", func_8014A548);
 
-INCLUDE_ASM(s32, "code_e0b30_len_b80", _set_music_track);
+s32 _set_music_track(s32 playerIndex, SongID songID, s32 variation, s32 fadeOutTime, s16 volume) {
+    GameStatus* gameStatus = GAME_STATUS;
+
+    if (gameStatus->demoState != 0) {
+        return 1;
+    } else {
+        // TODO: these two lines likely a MUSIC_PLAYER(idx) macro
+        MusicPlayer* musicPlayers = gMusicPlayers;
```

Comment:
my guess is that this pattern might produce the correct codegen but isn't quite correct. Probably they have two instances of this particular line, one in each scope or so, then just always use musicPlayers[idx] or so. Maybe I'm wrong, though